--- a/gfx/drivers/sdl_dingux_gfx.c	2026-09-10 20:45:47.540545495 +0800
+++ b/gfx/drivers/sdl_dingux_gfx.c	2026-09-10 20:45:47.540545495 +0800
@@ -423,7 +423,7 @@
    const char *input_drv_name                    = settings->arrays.input_driver;
    const char *joypad_drv_name                   = settings->arrays.input_joypad_driver;
    uint32_t surface_flags                        = (video->vsync)
-         ? (SDL_HWSURFACE | SDL_TRIPLEBUF | SDL_FULLSCREEN)
+         ? (SDL_HWSURFACE | SDL_DOUBLEBUF | SDL_FULLSCREEN)
          : (SDL_HWSURFACE | SDL_FULLSCREEN);
 
    /* Initialise graphics subsystem, if required */
@@ -614,7 +614,7 @@
    unsigned sanitized_width;
    unsigned sanitized_height;
    uint32_t surface_flags = (vid->vsync)
-         ? (SDL_HWSURFACE | SDL_TRIPLEBUF | SDL_FULLSCREEN)
+         ? (SDL_HWSURFACE | SDL_DOUBLEBUF | SDL_FULLSCREEN)
          : (SDL_HWSURFACE | SDL_FULLSCREEN);
 
    /* Cache set parameters */
